app/services/notification_service.rb

```diff
@@ -0,0 +1,64 @@
+class NotificationService
+  include Rails.application.routes.url_helpers
+
+  DEFAULT_LOCALE = :en
+
+  attr_reader :client
+
+  def initialize
+    @client = Twilio::REST::Client.new(twilio_account_sid, twilio_auth_token)
+  end
+
+  def send_sms(recipient_number, message)
+    sender_number    = twilio_sender_number
+    recipient_number = parse_phone_number(recipient_number)
+
+    send_twilio_message(sender_number, recipient_number, message)
+  end
+
+  def send_whatsapp(recipient_number, message)
+    sender_number    = "whatsapp:" + twilio_sender_number
+    recipient_number = "whatsapp:" + parse_phone_number(recipient_number)
+
+    send_twilio_message(sender_number, recipient_number, message)
+  end
+
+  def parse_phone_number(number)
+    parsed_number = Phonelib.parse(number, Rails.application.config.country[:abbreviation]).raw_national
+    default_country_code + parsed_number
+  end
+
+  private
+
+  def default_country_code
+    Rails.application.config.country[:sms_country_code]
+  end
+
+  def send_twilio_message(sender_number, recipient_number, message)
+    client.messages.create(
+      from: sender_number,
+      to: recipient_number,
+      status_callback: twilio_callback_url,
```

Comment:
We should probably not have `status_callback` for OTP SMSes, the Twilio callback is currently quite tied to `Communication` – so it should just be for reminders.

Reply:
Agree; fixed this so that callback URLs are optional.